### DISCUSSION TITLE
Unable to transfer using sendTransaction in IOS

### DISCUSSION THREAD

**Discussion by souravsoftwareer** [WalletConnect/walletconnect-monorepo#3403] [Developer Support]
I am getting issue in transfer. it is working fine in Android no issue at all.But in IOS transfer is not happening it gives following error
![1691561989231](https://github.com/WalletConnect/walletconnect-monorepo/assets/8158259/c77ca5f1-051e-41fe-92bb-361a145e95b5)

Uploading VID20230809115026.mp4…

**Comment by glitch-txs** [comment]
hi, which wallet is that? also could you share your code configuration or a code snippet to reproduce this issue?

**Comment by souravsoftwareer** [comment]
I am having issue in following example. It is working fine for android but not for IOS
https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer
Trust wallet version is 9.2

**Reply by glitch-txs** [reply]
can you share your code/ a minimal reproducible example?

**Comment by souravsoftwareer** [comment]
this issue is resolved but i have another issue i have tried to send MANA token but it is showing 0th over trust wallet i have send 9 from amount i am using following code for mana token

console.log("erc 20 coin")

    if (!web3Provider) {
      throw new Error('web3Provider not connected');
    }
    provider?.enable()

    const { chainId } = await web3Provider.getNetwork();
    if(chainId!=1) {
      await updateSessionMethod(chainId)
    }

    let web3 = new Web3(provider)
    let tokenAddress = item?.contract_address
    let minABI = transferErcAbi
    let decimals = item?.decimals ?? 18;

    let contract = new web3.eth.Contract(minABI, tokenAddress);
    console.log("decimals ==>", decimals)
    // let value = Web3Utils.toWei(amount, "ether")
    let value = sanitizeHex(numberToHex(parseFloat(amount)));
    
    let  value1 = ethers.utils.parseUnits(amount, decimals);

    contract.methods.transfer(walletAddress, Platform.OS == 'android'?value1:value)
      .send({ from: address })
      .on('transactionHash', async (resultt) => {

        // record a transaction
        console.log("result", resultt);
        sendEmailTransactionData(resultt, amount, item.tokenSymbol)
        await setAsyncItem(strings.OUT_FROM_HOME, "1")
        props.navigation.navigate('TransactionDetail', {
          txId: resultt,
          amount,
          destination: walletAddress,
          source: address,
          item: item
        })
      })
      .on('receipt', (receiptt) => {
        console.log("receipt", receiptt)
        // update transaction by hash
        let editObj = {
          trx_id: receiptt.transactionHash,
          status: 1,
          response_data: JSON.stringify(receiptt)

        }
      })
      .on('error', (err: any) => {

        setLoader(false)
        toast.error({ message: "Transaction Declined !!! Please try after some time" })
      })
      
      I have made video for it as well please check following link i have linked file as well in which transfer is working.
      You can check transferCoin() function i have used for.
      
      https://drive.google.com/file/d/1NUzW-cloVJEjftZjIDZDzgYzleG8wKvK/view?usp=sharing
      
      File link
      
      https://drive.google.com/file/d/1_ogJQU7mPNqdd6SKbBGhYuCfZpxwOieP/view?usp=sharing
